flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-maxcompute/src/main/java/org/apache/flink/cdc/connectors/maxcompute/coordinator/SessionManageOperator.java

```diff
@@ -0,0 +1,297 @@
+            Future<CoordinationResponse> waitForSuccess =
+                    submitRequestToOperator(new WaitForFlushSuccessRequest(indexOfThisSubtask));
+            output.collect(element);
+            // wait for sink writers flush success
+            waitForSuccess.get();
+            LOG.info(
+                    "operator {} handle FlushEvent end, all sink writers flush success",
+                    indexOfThisSubtask);
+        } else if (element.getValue() instanceof CreateTableEvent) {
+            TableId tableId = ((CreateTableEvent) element.getValue()).tableId();
+            Schema schema = ((CreateTableEvent) element.getValue()).getSchema();
+            schemaMaps.put(tableId, schema);
+            fieldGetterMaps.put(tableId, TypeConvertUtils.createFieldGetters(schema));
+            output.collect(element);
+        } else if (element.getValue() instanceof SchemaChangeEvent) {
+            SchemaChangeEvent schemaChangeEvent = (SchemaChangeEvent) element.getValue();
+            TableId tableId = schemaChangeEvent.tableId();
+            Schema newSchema =
+                    SchemaUtils.applySchemaChangeEvent(schemaMaps.get(tableId), schemaChangeEvent);
+            schemaMaps.put(tableId, newSchema);
+            fieldGetterMaps.put(tableId, TypeConvertUtils.createFieldGetters(newSchema));
+            output.collect(element);
+        } else {
+            output.collect(element);
+            LOG.warn("unknown element {}", element.getValue());
+        }
+    }
+
+    private void emitLatestSchema(TableId tableId) throws Exception {
+        Optional<Schema> schema = schemaEvolutionClient.getLatestSchema(tableId);
+        if (schema.isPresent()) {
+            Schema latestSchema = schema.get();
+            schemaMaps.put(tableId, latestSchema);
+            fieldGetterMaps.put(tableId, TypeConvertUtils.createFieldGetters(latestSchema));
+            output.collect(new StreamRecord<>(new CreateTableEvent(tableId, latestSchema)));
+        } else {
+            throw new RuntimeException(
+                    "Could not find schema message from SchemaRegistry for " + tableId);
+        }
+    }
+
+    @Override
+    public void prepareSnapshotPreBarrier(long checkpointId) throws Exception {
+        super.prepareSnapshotPreBarrier(checkpointId);
+        if (endOfInput) {
+            return;
+        }
+        LOG.info(
+                "operator {} prepare snapshot, wait for sink writers flush success",
+                indexOfThisSubtask);
+        // wait for sink writers flush success
+        waitLastSnapshotFlushSuccess();
+        snapshotFlushSuccess =
+                submitRequestToOperator(
+                        new WaitForFlushSuccessRequest(
+                                getRuntimeContext().getIndexOfThisSubtask()));
+    }
+
+    @Override
+    public void snapshotState(StateSnapshotContext context) throws Exception {
+        super.snapshotState(context);
+        sessionCache.clear();
```

Comment:
So, do we need to request a new session ID after each checkpoint, which may have a performance impact? It is expected that the checkpoint interval will need to be larger, right.

Reply:
Yes, we do. 
Each session can no longer be used after being committed, so we must re-create a session and request a new sessionID. 
And indeed, the checkpoint interval will need to be larger.